zcash_client_backend/src/tor/http/exchange_rate.rs

```diff
@@ -0,0 +1,373 @@
+use futures_util::join;
+use hyper::StatusCode;
+use iso_currency::Currency;
+use rand::{seq::IteratorRandom, thread_rng};
+use rust_decimal::Decimal;
+use serde::Deserialize;
+use tracing::{error, trace};
+
+use crate::tor::{Client, Error};
+
+impl Client {
+    /// Fetches the latest `currency`/ZEC exchange rate, derived from several exchanges.
+    ///
+    /// Supported currencies:
+    /// - USD
+    ///
+    /// Returns:
+    /// - `Ok(Some(rate))` if at least one exchange request succeeds.
+    /// - `Ok(None)` if the given currency is unsupported.
+    /// - `Err(_)` if none of the exchange queries succeed.
+    pub async fn get_exchange_rate(&self, currency: Currency) -> Result<Option<Decimal>, Error> {
+        let pair = match ExchangePair::get(currency) {
+            Some(pair) => pair,
+            None => return Ok(None),
+        };
+        assert!(matches!(pair, ExchangePair::Usd));
+
+        // Fetch the data in parallel.
+        let res = join!(
```

Comment:
I somewhat object that the user of this library has no option to exclude any of these data sources from consideration.